server/src/Repositories/genericRepository.js

```diff
@@ -4,7 +4,7 @@ const count = model => model.countDocuments();
 
 const findOne = async (model, data) => {
   try {
-    const result = await model.findOne({ _id: data }).lean();
+    const result = await model.findOne({ ...data });
```

Comment:
Why are you spreading here, when you can use entire data, the spread is used if you are cloning an object.